### PR TITLE
Fix undef/unused variable ESLint warnings across a number of files

### PR DESCRIPTION
**Pull request by greggraf**
This fixes all warnings.
(this work is towards issue: "Fix no-undef and no-unused-vars eslint violations" #238)

### CODE REVIEW

File: static/js/components/done.js [hunk 1]
```diff
@@ -1,14 +1,12 @@
 const html = require('choo/html');
-const find = require('lodash/find');
 const promote = require('./promote.js');
 const callcount = require('./callcount.js');
 
 
 module.exports = (state, prev, send) => {
-    const issue = find(state.issues, ['id', state.location.params.issueid]);
```
Comment (greggraf) [inline static/js/components/done.js:8]: this logic was moved to promote.js

**Comment by cdoremus** [comment]
@nickoneill I checked this PR and it looks good to merge into master, but before I do that I'd like to have @greggraf change ```no-undef``` and ```no-unused-vars``` rules from warning to error in the root ```.eslintrc.json``` file. This will assure that an future violations of the eslint rules fixed by this PR will cause CI failures. Are you OK with that change?

Anyone else who has an opinion about this, please chime in.

**Comment by nickoneill** [comment]
Yep! Coming from the fresh start of this PR means that failing on future undef or unused-vars is totally acceptable. Merge when you're happy with it @cdoremus!

**Comment by cdoremus** [comment]
@greggraf, change ```no-undef``` and ```no-unused-vars``` rules from warning to error in the root ```.eslintrc.json``` file and push them to this PR and I will be happy to approve it and merge it into master.

**Comment by greggraf** [comment]
@cdoremus - Thanks, great comment. I updated the PR.

**Review by cdoremus — APPROVED** [review]
It's good to go to master now. Good job!

**Pull request merged**